sigmastate/src/main/scala/sigmastate/basics/DLogProtocol.scala

```diff
@@ -110,7 +115,7 @@ object DLogProtocol {
   object DLogInteractiveProver {
     import CryptoConstants.secureRandom
 
-    def firstMessage(publicInput: ProveDlog): (BigInteger, FirstDLogProverMessage) = {
+    def firstMessage(): (BigInteger, FirstDLogProverMessage) = {
```

Comment:
Suggest rename `randomFirstMessage`

Reply:
@aslesarenko the name proposed by you is quite misleading. Please study the sigma protocols problem space language, the good introduction can be found in Chapter 6 of Lindell-Hazay book.

Reply:
@aslesarenko "randomFirstMessage" is simply wrong, as the parameter is not a sigma-protocol message and also not random

Reply:
@aslesarenko Please describe your logic when you suggest renamings

Reply:
`firstMessage` will produce different result on each invocation, since random generation happens inside. But is sounds like a pure function. I didn't noticed it before, since there was the parameter declared (but not actually used). 
Now, randomness side effect is hidden under the "pure sounding name".

Reply:
@aslesarenko we do not have naming rules for pure/unpure functions, and how's that related to proposed PARAMETER rename?

Reply:
Not sure which parameter renaming you are talking about. I see you removed the parameter. 
I proposed method rename.

Reply:
@aslesarenko if your suggestion was about the function name (please clearly describe what and why you're suggesting to rename further), than we're not separating this around all the codebase, also, it is not common in the cryptography-related code in general (as cryptography is randomized).  Haskell libs are wrapping function results into special monads , but that's quite different story.